### PR TITLE
[AMD] Fix DeepSeekV4 server cutlass error

### PR DESCRIPTION
**Pull request by At1a8**
## Motivation

PR #25220 introduced a top-level import of the CuTeDSL MQA logits module, which depends on NVIDIA-only cutlass packages. 

On AMD/ROCm environments without cutlass, this import fails during DSV4 model registration and prevents the server from starting, even though the CuTeDSL backend is not used.
image

## Modifications

Guard the CuTeDSL imports behind `not is_hip()`. On non-ROCm platforms, preserve the original eager import behavior so NVIDIA/CuTeDSL logic remains unchanged.

## Accuracy Tests

root@smci355-ccs-aus-n12-13:/sgl-workspace/sglang# python3 /sgl-workspace/sglang/benchmark/gsm8k/bench_sglang.py --port 8000 --num-questions 1319

100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1319/1319 [01:08<00:00, 19.29it/s]
Accuracy: 0.948
Invalid: 0.000
Latency: 68.365 s
Output throughput: 1728.667 token/s

## Speed Tests and Profiling
This PR has almost no impact on performance.
TP8 DP8, 8k/1k, concurrency = 256, num prompts = 4 × concurrency | Total token throughput (tok/s) | Mean TTFT (ms) | Mean TPOT (ms)
-- | -- | -- | --
Main branch | failed | failed | failed 
This PR | 34904.71 | 6871.52 | 57.17

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: [Run #28877915192](https://github.com/sgl-project/sglang/actions/runs/28877915192)
Latest PR Test (Extra): :x: [Run #28877913869](https://github.com/sgl-project/sglang/actions/runs/28877913869)

### CODE REVIEW

File: python/sglang/srt/layers/attention/dsa/dsa_indexer.py [hunk 1]
```diff
@@ -65,6 +64,9 @@
 _is_cuda = is_cuda()
 _is_hip = is_hip()
 _is_npu = is_npu()
+if not _is_hip:
+    # Preserve the original eager import behavior on non-ROCm platforms.
+    from sglang.jit_kernel.dsa import pick_dsl_expand
```
[the remark this answers is not in the document]
Comment (kkHuang-amd) [reply python/sglang/srt/layers/attention/dsa/dsa_indexer.py:69]: Please check this review comment and fix that

**Review by b8zhong — APPROVED** [review]
Thanks. Sorry about the breaking

**Comment by At1a8** [comment]
image
DeepSeek V4 passed in local terminal.

File: python/sglang/srt/layers/attention/dsa/dsa_indexer.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (At1a8) [reply python/sglang/srt/layers/attention/dsa/dsa_indexer.py:69]: > Please check this review comment and fix that

Thanks for your comments! I've addressed the issue based on your suggestions.
image

**Review by HaiShaw — APPROVED** [review]
